decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/data/BasesMask.java

```diff
@@ -0,0 +1,310 @@
+            if (i == 0) {
+                return null;
+            } else {
+                return Integer.parseInt(length);
+            }
+        } else {
+            throw new IllegalArgumentException("Input value [" + length + "] is not supported");
+        }
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getVal("y", readOneIncludeLength));
+        sb.append(getVal("n", readOneIgnoreLength));
+        sb.append(",");
+        sb.append(getVal("i", indexOneIncludeLength));
+        sb.append(getVal("n", indexOneIgnoreLength));
+        sb.append(",");
+        if (indexTwoIncludeLength != null || indexTwoIgnoreLength != null) {
+            sb.append(getVal("i", indexTwoIncludeLength));
+            sb.append(getVal("n", indexTwoIgnoreLength));
+            sb.append(",");
+        }
+        sb.append(getVal("y", readTwoIncludeLength));
+        sb.append(getVal("n", readTwoIgnoreLength));
+        return sb.toString();
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 5;
+        hash = 83 * hash + Objects.hashCode(this.readOneIncludeLength);
+        hash = 83 * hash + Objects.hashCode(this.readOneIgnoreLength);
+        hash = 83 * hash + Objects.hashCode(this.indexOneIncludeLength);
+        hash = 83 * hash + Objects.hashCode(this.indexOneIgnoreLength);
+        hash = 83 * hash + Objects.hashCode(this.indexTwoIncludeLength);
+        hash = 83 * hash + Objects.hashCode(this.indexTwoIgnoreLength);
+        hash = 83 * hash + Objects.hashCode(this.readTwoIncludeLength);
+        hash = 83 * hash + Objects.hashCode(this.readTwoIgnoreLength);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final BasesMask other = (BasesMask) obj;
+        if (!Objects.equals(this.readOneIncludeLength, other.readOneIncludeLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.readOneIgnoreLength, other.readOneIgnoreLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.indexOneIncludeLength, other.indexOneIncludeLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.indexOneIgnoreLength, other.indexOneIgnoreLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.indexTwoIncludeLength, other.indexTwoIncludeLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.indexTwoIgnoreLength, other.indexTwoIgnoreLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.readTwoIncludeLength, other.readTwoIncludeLength)) {
+            return false;
+        }
+        if (!Objects.equals(this.readTwoIgnoreLength, other.readTwoIgnoreLength)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)